Add response headers to RequestLog

kaapstorm opened the pull request:
## Summary

When I was looking at RequestLog today, I noticed how easy it would be to log response headers.

I've been wanting to do this, because DHIS2 passes ETag values in its response headers. We use these ETag values to avoid conflicts when pushing data. So logging them is useful.

![image](https://user-images.githubusercontent.com/708421/105400306-6bfda200-5c2d-11eb-85b3-9dfe990ed92d.png)

:fish: :blowfish: :tropical_fish: Probably easier to review by commit.


## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression for the reasons below


### Safety story

Minor change.

**NOTE:** Includes a migration to add a column to a table that is relatively large on Prod. The transaction should be fast: Values aren't backfilled and the column is not indexed. But I think that rolling back the migration (i.e. dropping the column) would be slower.

Migration and rollback both tested locally.

[**EDIT:** 2021-03-01] Migration already performed on Prod and Staging using limited-release deploys.


### Rollback instructions

The migration in this PR will need to be rolled back if the code is reverted, but rolling back the migration would not have to be done immediately. The migration adds a column to a table, so the reverted codebase will just ignore it. The migration can be rolled back later, after the reverted PR is deployed, as follows:

1. Do a private release deploy of this branch ("nh/resp_head").
2. Log into the "django_manage[0]" host
3. Activate the virtualenv in the private release directory
4. Migrate the "motech" app back to its previous state:
   ```bash
       $ ./manage.py migrate motech 0007
   ```

kaapstorm committed:
♻ Move RequestLogEntry class

kaapstorm committed:
Add response_headers to RequestLogEntry

Important when API uses ETag

kaapstorm committed:
Add response_headers to RequestLog

kaapstorm committed:
Render response headers in detail template

kaapstorm committed:
Update tests

proteusvacuum commented on corehq/motech/templates/motech/log_detail.html:
```diff
@@ -50,6 +50,12 @@
+          <tr>
+            <th>{% trans "Response headers" %}:</th>
+            <td>{% if log.response_headers %}
+              <pre>{{ log.response_headers|pp_json }}</pre>{% else %}&nbsp;
```
Might be weird for people looking at their old logs to see this be blank. Maybe add a note that this isn't always populated?

kaapstorm committed:
Merge branch 'master' into nh/resp_head

kaapstorm commented:
This migration locks the table for PostgreSQL<11, because it has a non-null default value.

Changing the PR to draft. I will make the default null, recreate the migration, and test.

kaapstorm committed:
Use nullable field for a non-locking migration

Before (locks table on PostgreSQL < 11):
    ALTER TABLE "dhis2_jsonapilog" ADD COLUMN "response_headers" text DEFAULT '""' NOT NULL

After (does not lock table):
    ALTER TABLE "dhis2_jsonapilog" ADD COLUMN "response_headers" text NULL

kaapstorm commented:
The new migration works fine. I have migrated Staging and Prod using a limited-release deploys.
